custom_components/tuya_local/devices/hombli_thermostatic_radiator_valve.yaml

```diff
@@ -0,0 +1,58 @@
+name: Hombli Thermostatic Radiator Valve
+products:
+  - id: ngxyxyc8
+    name: Hombli HBRT-0109
+primary_entity:
+  entity: climate
+  dps:
+    - id: 8
+      type: boolean
+      name: window_check
+      icon: "mdi:window-open"
+    - id: 10
+      type: boolean
+      name: frost
+      mapping:
+        - dps_val: true
+          value: "On"
+        - dps_val: false
+          value: "Off"
+    - id: 27
+      type: integer
+      name: temp_correction
+      mapping:
+        - scale: 10
+    - id: 40
+      type: boolean
+      name: child_lock
+    - id: 101
+      type: boolean
+      name: hvac_mode
+      mapping:
+        - dps_val: false
+          value: "off"
+        - dps_val: true
+          value: "auto"
```

Comment:
I would add a condition here on the program dp, so you can switch that through hvac_mode as well:
```
      - dps_val: true
        constraint: program
        conditions:
           - dps_val: true
             value: auto
           - dps_val: false
             value: heat
```

Reply:
I've been looking into this an kind of disagree on the logic. The TRV has modes like this:

off: valve closed permanently
on: valve open always
auto: value open/closed according to set temperature threshold 
program: mode off/on/auto according to programmed schedule

I think 'heat' mode doesn't make a lot of sense as a mode, since it's actually a state of "valve open".

Reply:
Away mode is program=off + set temp = 16 as well

Reply:
- off: valve closed permanently **= HVACMode.OFF**
- on: valve open always *HA Climate platform has no equivalent*
- auto: value open/closed according to set temperature threshold **= HVACMode.HEAT**
- program: mode off/on/auto according to programmed schedule **= HVACMode.AUTO**

Reply:
I guess the options then are to implement dp 101 as switching between "off" and "heat", and implement "auto" and "program" by separate switches (I only actually see the switch for program though).  But I'm not sure of the behaviour of dp 101 in the case that program is enabled. Does it become a sensor that indicates the actual state of the valve? Or is it required to be on (dp 101 overrides dp 108), or does the state not matter (dp 108 overrides dp 101)?